kubernetes/deployments_kubeclient.go

```diff
@@ -0,0 +1,1339 @@
+			}
+		}
+		if match {
+			rcsForDc = append(rcsForDc, rc)
+		}
+	}
+
+	return rcsForDc, nil
+}
+
+func (kc *kubeClient) getResourceQuota(namespace string) (*app.EnvStats, error) {
+	const computeResources string = "compute-resources"
+	quota, err := kc.ResourceQuotas(namespace).Get(computeResources, metaV1.GetOptions{})
+	if err != nil {
+		return nil, errs.WithStack(err)
+	} else if quota == nil {
+		return nil, errs.Errorf("no resource quota with name: %s", computeResources)
+	}
+
+	// Convert quantities to floating point, as this should provide enough
+	// precision in practice
+	cpuLimit, err := quantityToFloat64(quota.Status.Hard[v1.ResourceLimitsCPU])
+	if err != nil {
+		return nil, errs.WithStack(err)
+	}
+	cpuUsed, err := quantityToFloat64(quota.Status.Used[v1.ResourceLimitsCPU])
+	if err != nil {
+		return nil, errs.WithStack(err)
+	}
+
+	cpuStats := &app.EnvStatCores{
+		Quota: &cpuLimit,
+		Used:  &cpuUsed,
+	}
+
+	memLimit, err := quantityToFloat64(quota.Status.Hard[v1.ResourceLimitsMemory])
+	if err != nil {
+		return nil, errs.WithStack(err)
+	}
+
+	memUsed, err := quantityToFloat64(quota.Status.Used[v1.ResourceLimitsMemory])
+	if err != nil {
+		return nil, errs.WithStack(err)
+	}
+
+	memUnits := "bytes"
+	memStats := &app.EnvStatMemory{
+		Quota: &memLimit,
+		Used:  &memUsed,
+		Units: &memUnits,
+	}
+
+	result := &app.EnvStats{
+		Cpucores: cpuStats,
+		Memory:   memStats,
+	}
+
+	return result, nil
+}
+
+func quantityToFloat64(q resource.Quantity) (float64, error) {
+	val64, rc := q.AsInt64()
+	var result float64
+	if rc {
+		result = float64(val64)
+	} else {
+		valDec := q.AsDec()
+		val64, ok := valDec.Unscaled()
+		if !ok {
+			return -1, errs.Errorf("%s cannot be represented as a 64-bit integer", valDec.String())
+		}
+		// From dec.go: The mathematical value of a Dec equals: unscaled * 10**(-scale)
+		result = float64(val64) * math.Pow10(-int(valDec.Scale()))
+	}
+	return result, nil
+}
+
+// GetPodsInNamespace - return all pods in namepsace 'nameSpace' and application 'appName'
+func (kc *kubeClient) GetPodsInNamespace(nameSpace string, appName string) ([]v1.Pod, error) {
+	listOptions := metaV1.ListOptions{
+		LabelSelector: "app=" + appName,
+	}
+	pods, err := kc.Pods(nameSpace).List(listOptions)
+	if err != nil {
+		return nil, errs.WithStack(err)
+	}
+	return pods.Items, nil
+}
+
+func (kc *kubeClient) getPods(namespace string, uid types.UID) ([]*v1.Pod, error) {
+	pods, err := kc.Pods(namespace).List(metaV1.ListOptions{})
+	if err != nil {
+		return nil, errs.WithStack(err)
+	}
+
+	appPods := []*v1.Pod{}
+	for _, pod := range pods.Items {
+		// If a pod belongs to a given RC, it should have an OwnerReference
+		// whose UID matches that of the RC
+		// https://github.com/openshift/origin-web-console/blob/v3.7.0/app/scripts/services/ownerReferences.js#L40
+		match := false
+		for _, ref := range pod.OwnerReferences {
+			if ref.UID == uid && ref.Controller != nil && *ref.Controller {
+				match = true
```

Comment:
should you add a `break` to stop iterating over the `pod.OwnerReferences` ?

Reply:
fixed.

Reply:
ok